tests/main/parallel-install-auto-aliases/task.yaml

```diff
@@ -87,3 +87,43 @@ execute: |
     # aliases are unchanged
     test "$(readlink "$SNAP_MOUNT_DIR"/bin/test_snapd_wellknown1)" = "test-snapd-auto-aliases.wellknown1"
     test "$(readlink "$SNAP_MOUNT_DIR"/bin/test_snapd_wellknown2)" = "test-snapd-auto-aliases.wellknown2"
+
+    echo "Installing test-snapd-auto-aliases_bar with --prefer"
+    snap install --prefer test-snapd-auto-aliases_bar
+
+    echo "Aliases should belong to test-snapd-auto-aliases_bar"
+    snap aliases > aliases.out
+    MATCH "test-snapd-auto-aliases_bar.wellknown1 +test_snapd_wellknown1 +-"        < aliases.out
+    MATCH "test-snapd-auto-aliases_bar.wellknown2 +test_snapd_wellknown2 +-"        < aliases.out
+    MATCH "test-snapd-auto-aliases_foo.wellknown1 +test_snapd_wellknown1 +disabled" < aliases.out
+    MATCH "test-snapd-auto-aliases_foo.wellknown2 +test_snapd_wellknown2 +disabled" < aliases.out
+    MATCH "test-snapd-auto-aliases.wellknown1 +test_snapd_wellknown1 +disabled"     < aliases.out
+    MATCH "test-snapd-auto-aliases.wellknown2 +test_snapd_wellknown2 +disabled"     < aliases.out
+
+    # aliases belong to test-snapd-auto-aliases_bar
+    test "$(readlink "$SNAP_MOUNT_DIR"/bin/test_snapd_wellknown1)" = "test-snapd-auto-aliases_bar.wellknown1"
+    test "$(readlink "$SNAP_MOUNT_DIR"/bin/test_snapd_wellknown2)" = "test-snapd-auto-aliases_bar.wellknown2"
+
+    # remove and install main snap with --prefer
```

Comment:
personal preference: would rather refer to main snap as `test-snapd-auto-aliases`